q2_sample_classifier/tests/test_visualization.py

```diff
@@ -137,6 +137,21 @@ def test_confusion_matrix_vmin_too_high_and_vmax_too_low(self):
                                     r'greater than.*\s\s0\.5.*less.*1\.0'):
             confusion_matrix(self.tmpd, self.a, b, vmin=.5, vmax=.5)
 
+    def test_confusion_matrix_dtype_coercion(self):
+        predictions = pd.Series([1, 1, 1, 2, 2, 2],
+                                index=pd.Index([i for i in 'abcdef'],
+                                name='sample_id'), name='features')
+
+        # NOTE: the targets are numbers but represented as str
+        truth = qiime2.CategoricalMetadataColumn(pd.Series(
+            [x for x in '121212'], index=pd.Index(
+                [i for i in 'abcdef'], name='sample-id'), name='target'))
+
+        confusion_matrix(self.tmpd, predictions, truth)
+
+        # checkpoint assertion
+        self.assertTrue(True)
```

Comment:
Is there anything more concrete we can test here?

Reply:
I can't really think of anything. My understanding is that we just want to make sure the `confusion_matrix` was created.

Reply:
So then let's do that! Are there files or assets from the viz you can verify?

Reply:
I have no idea. The other tests that are meant to actually create a viz without errors don't assert anything at all, and I can't inspect the `confusion_matrix` that gets created right now because the test explodes in pandas before the matrix is created. Presumably it will be putting something the viz into `self.tmpd` when it actually works, but I'm not sure what I should be doing with it in there, or how to assert that the specific viz I want exists.

Reply:
I think @thermokarst is suggesting that you just verify that the index.html exists, maybe verify that some contents of the file exist (but I will let @thermokarst specify if he wants more than verifying that the file exists). See an example here:
https://github.com/qiime2/q2-longitudinal/blob/19241ad5740f344579ca7b4914c2345a74c8f60a/q2_longitudinal/tests/test_longitudinal.py#L549-L566

Reply:
Awesome, thanks guys.